src/Language/Bitcoin/Script/Utils.hs

```diff
@@ -33,7 +33,9 @@ toCScriptNum n
     b = BS.snoc b' msb
 
 pushNumber :: Int -> ScriptOp
-pushNumber = opPushData . toCScriptNum
+pushNumber i
+    | i <= 16 = intToScriptOp i
```

Comment:
This seems very curious to me. Would you never `OP_PUSHDATA` on a number `<= 16`? Maybe I just don't know.

Reply:
Mixing op codes in this function seems strange is really what I'm saying.

Reply:
The goal here is to push bytes to the stack that a later opcode will interpret as a number.  The `OP_i` opcodes take one byte to push their number, versus the PUSHDATA opcodes, which require at least 2 bytes even for small numbers.